src/lando/api/tests/conftest.py

```diff
@@ -432,6 +432,28 @@ def hg_clone(hg_server, tmpdir):
     return clone_dir
 
 
+@pytest.fixture
+def git_repo(tmpdir):
```

Comment:
Would be nice to add a docstring here. It might be also more useful to add a path parameter to this fixture, so that you can pass it elsewhere as an optional.

Reply:
> Would be nice to add a docstring here. 

👍🏼 

I'll also move this to src.main.tests, to follow the move of  `test_git.py`.

I took the opportunity to clean up _those_ tests to use the `tmp_path: pathlib.Path` rather than `tmpdir: py.path` as we previously discussed. As the new test code is better separated,` it's easier to use the new pattern without impacting intra-file consistency of the legacy tests.

---

>  It might be also more useful to add a path parameter to this fixture, so that you can pass it elsewhere as an optional.

I'm not sure how to do it. I took a punt on something like this for the `conftest`

```
@pytest.fixture                                                                                                                                                                                                                               
def git_repo(tmp_path: pathlib.Path, repo_path: Optional[pathlib.Path] = None):
    repo_dir = tmp_path / "git_repo"
    if repo_path:
        repo_dir = repo_path       
  ....
```

and this in the test

``` 
@pytest.fixture
def repo_path():
    return Path("/non-existent-path")

def test_GitSCM_clone(tmp_path: Path, git_repo: Path, repo_path):
    ...
```

But that didn't seem to make any difference. Do you have an example of what it should look like here?

Reply:
A simple example of this maybe is the `normal_patch` fixture.

Reply:
Yup, I just realised that when looking at the test for #182 (: